pkg/operator/controller/controller.go

```diff
@@ -351,18 +351,28 @@ func (r *reconciler) ensureDNS(dns *operatorv1.DNS) error {
 		if _, _, err := r.ensureDNSConfigMap(dns, clusterDomain); err != nil {
 			errs = append(errs, fmt.Errorf("failed to create configmap for dns %s: %v", dns.Name, err))
 		}
+		endpoints := &corev1.Endpoints{}
 		if haveSvc, svc, err := r.ensureDNSService(dns, clusterIP, daemonsetRef); err != nil {
 			// Set clusterIP to an empty string to cause ClusterOperator to report
 			// Available=False and Degraded=True.
 			clusterIP = ""
 			errs = append(errs, fmt.Errorf("failed to create service for dns %s: %v", dns.Name, err))
 		} else if !haveSvc {
 			errs = append(errs, fmt.Errorf("failed to get service for dns %s", dns.Name))
-		} else if err := r.ensureMetricsIntegration(dns, svc, daemonsetRef); err != nil {
-			errs = append(errs, fmt.Errorf("failed to integrate metrics with openshift-monitoring for dns %s: %v", dns.Name, err))
+		} else {
+			if err := r.ensureMetricsIntegration(dns, svc, daemonsetRef); err != nil {
+				errs = append(errs, fmt.Errorf("failed to integrate metrics with openshift-monitoring for dns %s: %v", dns.Name, err))
+			}
+			name := types.NamespacedName{
```

Comment:
nit: Add some comments here to make it clear that the metrics integration and endpoint GET call are not related.

Reply:
Done.